torch/_functorch/aot_autograd.py

```diff
@@ -2415,6 +2432,7 @@ def create_synthetic_base_metadata(
             # mutations, they will be hidden from the rest of aot autograd.
             mutates_data=True if len(outer_indices) > 1 else m.input_info[outer_indices[0]].mutates_data,
             mutates_metadata=False if len(outer_indices) > 1 else m.input_info[outer_indices[0]].mutates_metadata,
+            mutations_hidden_from_autograd=all(m.input_info[x].mutations_hidden_from_autograd for x in outer_indices),
```

Comment:
We agreed offline that:

(1) one thing missing from this PR is that AOTAutograd should use this new information to decide to unconditionally plumb "triton-only input mutations" directly from inductor. Instead, today it will sometimes hide these input mutation in a runtime epilogue

(2) This is a perf issues and not a correctness issue - good to do in a followup and not necessary for this PR.